### ISSUE TITLE
Undefined variable error is not correctly reported

### ISSUE THREAD

**Issue by pditommaso** [nextflow-io/nextflow#780]
## Bug report 

When the command wrapper evaluates an undefined variable the execution stops and the `.command.out` and `.command.err` files are not created. However the exit status reported is `0`, therefore the resulting error message is misleading making difficult to debug the task 

### Expected behavior and actual behavior

The task should return a non-zero error message

### Program output 

```
Caused by:
  Missing 'stdout' file: /.../.command.out for process > foo
``` 
### Steps to reproduce the problem

```
process foo {
  """
    echo ciao
  """
}
```

nextflow.config
```
env.X='$Y'
```

### Environment 

* Nextflow version: 0.30.2
* Java version: any
* Operating system: any

**Comment by pditommaso** [comment]
This is caused by a weird Bash behaviour when when defining `set -e` in the script. See for details [here](https://stackoverflow.com/questions/51154169/why-bash-trap-does-not-capture-exit-status-for-an-undefined-variable?noredirect=1#comment89291657_51154169). 


Possible solution to capture the error condition is to use `set -E` instead of `set -e` adding an `ERR` trap 

```
set -E
set -u

on_exit() {
  exit_status=${ret:=$?}
  echo exit_status=$exit_status
  exit $exit_status
}

on_error() {
  exit_status=${ret:=$?}
  exit $exit_status
}

trap on_exit EXIT
trap on_error ERR

X=$Y

echo done
```

**Comment by pditommaso** [comment]
It turns out that is Bash bug fixed in version 4.1 and later. See [here](https://stackoverflow.com/questions/51154169/why-does-bash-trap-not-capture-the-exit-status-for-an-undefined-variable?noredirect=1#comment89291657_51154169) for details.
